### ISSUE TITLE
`HyperparameterTuner.attach()` does not get `use_spot_instances` setting

### ISSUE THREAD

**Issue by dz902** [aws/sagemaker-python-sdk#1817]
**Describe the bug**

When using the following:

```
PARENT_TUNER = HyperparameterTuner.attach(
    tuning_job_name = PARENT_TUNING_JOB_NAME
)
```

...on a tuning job where its job definition has:

```
...
        "StoppingCondition": {
            "MaxRuntimeInSeconds": 3600,
            "MaxWaitTimeInSeconds": 7200
        },
        "EnableNetworkIsolation": false,
        "EnableInterContainerTrafficEncryption": false,
        "EnableManagedSpotTraining": true
...
```

The `max_wait` and `use_spot_instances` setting are both `None`. I traced back to:

https://github.com/aws/sagemaker-python-sdk/blob/481719f90f7d9bbf81de4dd53bc418991945b866/src/sagemaker/estimator.py#L811-L874

It seems `use_spot_instances` and `max_wait` do not get carried over to newly created estimator.

**To reproduce**

See above.

**Expected behavior**

`use_spot_instances` and `max_wait` etc. should all be carried over to newly `attach()`ed tuner. This also affects warm start helpers like `identical_data_and_algorithm()`.

```**Screenshots or logs**
If applicable, add screenshots or logs to help explain your problem.

**System information**
A description of your system. Please provide:
- **SageMaker Python SDK version**: v2.0.0
- **Framework name (eg. PyTorch) or algorithm (eg. KMeans)**:
- **Framework version**:
- **Python version**:
- **CPU or GPU**:
- **Custom Docker image (Y/N)**: N, official image classification image

**Additional context**
Add any other context about the problem here.

**Comment by dz902** [comment]
Also `inputs` from parent tuner is ignored in newly created tuner when calling `fit()` without the `inputs` arg. In official docs `fit()` was called with input as an empty set `input={}` or just plain `fit()`, which to me indicates that there is a way to inherit of set `inputs`, but it doesn't seem to be the case.

**Comment by dz902** [comment]
Until this is fixed, a workaround:

```

PARENT_TUNER.estimator.max_wait = 7200 # !!!
PARENT_TUNER.estimator.use_spot_instances = True # !!!

```

**Comment by icywang86rui** [comment]
The fix is merged and released in v2.5.5. @dz902 thanks for reporting the bug!
